src/Ssh/Ssh/Common/BastionUtils.cs

```diff
@@ -0,0 +1,318 @@
+
+    {
+    internal class BastionUtils
+    {
+        #region Fields
+        private NetworkClient _networkClient;
+        private IAzureContext _context;
+        #endregion
+
+        public BastionUtils(IAzureContext context)
+        {
+            _context = context;
+        }
+
+        #region Properties
+        private NetworkClient NetworkClient
+        {
+            get
+            {
+                if (_networkClient == null)
+                {
+                    _networkClient = new NetworkClient(_context);
+                }
+                return _networkClient;
+            }
+
+            set { _networkClient = value; }
+        }
+        public IBastionHostsOperations BastionClient
+        {
+            get
+            {
+                return NetworkClient.NetworkManagementClient.BastionHosts;
+            }
+        }
+        #endregion
+        public int HandleBastionProperties(NetworkInterface nic, string vmName, IAzureContext context, string vmPort, Process sshProcess)
+        {
+            int port = 22;
+            string location = nic.Location;
+            CheckValidBastionDeveloperLocation(location);
+
+            string vmSubscriptionID = nic.VirtualMachine.Id;
+
+
+            string vNetId = null;
+            string vNetName = null;
+            string vNetResourceGroup = null;
+            if (nic.IpConfigurations != null && nic.IpConfigurations.Any())
+            {
+                foreach (var ipConfig in nic.IpConfigurations)
+                {
+                    if (ipConfig.Subnet != null)
+                    {
+                        string subnetId = ipConfig.Subnet.Id;
+
+                        (vNetId, vNetName, vNetResourceGroup) = GetVNetDetailsFromSubnetId(subnetId);
+                        break;
+                    }
+                }
+            }
+            ResourceGraphUtils resourceGraphUtils = new ResourceGraphUtils(context);
+            string bastionsFoundInVNet = resourceGraphUtils.QueryResourceGraph(vNetId);
+            string bastionNameInVNet = ParseAvailableBastions(bastionsFoundInVNet);
+
+            BastionHost bastion;
+
+
+            if (bastionNameInVNet == null)
+            {
+                string bastionName = vNetName + "-bastion";
+                bastion = CreateDeveloperBastion(vNetResourceGroup, bastionName, location, vNetId);
+
+            }
+            else
+            {
+                bastion = FetchDeveloperBastion(vNetResourceGroup, bastionNameInVNet);
+
+            }
+            int bastionPort = 0;  
+            string bastionEndPoint = null;
+            try
+            {
+
+
+                bastionEndPoint = GetDataPodEndPoint(bastion, context, vmSubscriptionID, port);
+
+            }
+            catch (Exception ex)
+            {
+                string error = "Error fetching end point: " + ex.Message;
+                throw new AzPSCloudException(error);
```

Comment:
You can use string interpolation:

```suggestion
                throw new AzPSCloudException($"Error fetching end point: {ex.Message}");
```